### PR TITLE
Corrected NodeIterator:whatToShow example to use bit values

### PR DESCRIPTION
**Pull request by jackdeguest**
Corrected the example to use bit values and bit AND operator rather than addition.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [X] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/nodeiterator/whattoshow/index.md [hunk 1]
```diff
@@ -123,12 +123,12 @@ The values that can be combined to form the bitmask are:
 ```js
 var nodeIterator = document.createNodeIterator(
     document.body,
-    NodeFilter.SHOW_ELEMENT + NodeFilter.SHOW_COMMENT + NodeFilter.SHOW_TEXT,
+    ( NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT ),
     { acceptNode: function(node) { return NodeFilter.FILTER_ACCEPT; } },
     false
 );
 if( (nodeIterator.whatToShow == NodeFilter.SHOW_ALL) ||
```
Comment (sideshowbarker) [inline files/en-us/web/api/nodeiterator/whattoshow/index.md:130]:
```suggestion
if( (nodeIterator.whatToShow & NodeFilter.SHOW_ALL) ||
```
As long as we’re fixing this example overall, wouldn’t it be more idiomatic to use a bitwise AND for the test here too?

File: files/en-us/web/api/nodeiterator/whattoshow/index.md [hunk 2]
```diff
@@ -123,12 +123,12 @@ The values that can be combined to form the bitmask are:
...
-    (nodeIterator.whatToShow % (NodeFilter.SHOW_COMMENT*2)) >= NodeFilter.SHOW_COMMENT) {
+    (nodeIterator.whatToShow & NodeFilter.SHOW_COMMENT ) > 0 ) {
```
Comment (sideshowbarker) [inline files/en-us/web/api/nodeiterator/whattoshow/index.md:131]:
```suggestion
    (nodeIterator.whatToShow & NodeFilter.SHOW_COMMENT)) {
```
Wouldn’t it be more idiomatic to only rely on the truthiness/falsiness of the bitwise operation itself? — rather than doing an additional explicit comparison to 0?

(And anyway if we were to do an additional comparison here, it seems like it’d be more instructive to just precisely check either that it’s equal to 1 or that it’s not zero — since those are the only possible values)

Comment (jackdeguest) [reply files/en-us/web/api/nodeiterator/whattoshow/index.md:131]: If you set, for example, `whatToShow` to `NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT` and then later do the check `nodeIterator.whatToShow & NodeFilter.SHOW_COMMENT` you would get 128, not 0. You would get 0 if the bitwise comparison failed.

Comment (sideshowbarker) [reply files/en-us/web/api/nodeiterator/whattoshow/index.md:131]: > If you set, for example, `whatToShow` to `NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT` and then later do the check `nodeIterator.whatToShow & NodeFilter.SHOW_COMMENT` you would get 128, not 0.

Sorry right yeah, I misspoke about it only being 1 or 0

> You would get 0 if the bitwise comparison failed.

Yeah that’s the point I had actually been wanting to make in my comment: to make the test evaluate just the bitwise comparison itself, without explicitly comparing it to 0 — which seems unnecessary, and less idiomatic.

So in other words, do you think we would lose anything by just doing the following?

```js
if ((nodeIterator.whatToShow & NodeFilter.SHOW_ALL) ||
    (nodeIterator.whatToShow & NodeFilter.SHOW_COMMENT)) { … }
```

Comment (jackdeguest) [reply files/en-us/web/api/nodeiterator/whattoshow/index.md:131]: No of course. I agree with you.

File: files/en-us/web/api/nodeiterator/whattoshow/index.md [hunk 3]
```diff
@@ -123,12 +123,12 @@ The values that can be combined to form the bitmask are:
...
 var nodeIterator = document.createNodeIterator(
     document.body,
-    NodeFilter.SHOW_ELEMENT + NodeFilter.SHOW_COMMENT + NodeFilter.SHOW_TEXT,
+    ( NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT ),
     { acceptNode: function(node) { return NodeFilter.FILTER_ACCEPT; } },
     false
 );
-if( (nodeIterator.whatToShow == NodeFilter.SHOW_ALL) ||
-    (nodeIterator.whatToShow % (NodeFilter.SHOW_COMMENT*2)) >= NodeFilter.SHOW_COMMENT) {
+if( (nodeIterator.whatToShow & NodeFilter.SHOW_ALL) ||
```
Comment (sideshowbarker) [inline files/en-us/web/api/nodeiterator/whattoshow/index.md:130]:
```suggestion
if ((nodeIterator.whatToShow & NodeFilter.SHOW_ALL) ||
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
Thanks much for fixing this up. As messed up and off the mark as this example was before you fixed it, I guess it could only have crept in back in the wiki days when people could commit changes directly without them being reviewed first. Running into these cases always leaves me feeling worried about how many more cases like this one was that we might still have hanging around. But I guess the only thing we can do is just to fix them whenever we come across them.

On that note, I notice https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker/whatToShow#example has the same whacked example that you fixed in this PR.

Can you raise a PR with a fix for that article too?
